packages/kbn-text-based-editor/src/text_based_languages_editor.tsx

```diff
@@ -1107,6 +1138,65 @@ export const TextBasedLanguagesEditor = memo(function TextBasedLanguagesEditor({
           editorIsInline={editorIsInline}
         />
       )}
+
+      {createPortal(
+        Object.keys(popoverPosition).length !== 0 && popoverPosition.constructor === Object && (
+          <div
+            tabIndex={0}
+            style={{
+              ...popoverPosition,
+              backgroundColor: euiTheme.colors.emptyShade,
+              borderRadius: euiTheme.border.radius.small,
+              position: 'absolute',
+              overflow: 'auto',
+            }}
+            ref={popoverRef}
+            data-test-subj="TextBasedLangEditor-timepicker-popover"
+          >
+            <EuiDatePicker
+              selected={timePickerDate}
+              autoFocus
+              onChange={(date) => {
+                if (date) {
+                  setTimePickerDate(date);
+                }
+              }}
+              onSelect={(date, event) => {
+                if (date && event) {
+                  const currentCursorPosition = editor1.current?.getPosition();
+                  const lineContent = editorModel.current?.getLineContent(
+                    currentCursorPosition?.lineNumber ?? 0
+                  );
+                  const contentAfterCursor = lineContent?.substring(
+                    (currentCursorPosition?.column ?? 0) - 1,
+                    lineContent.length + 1
+                  );
+
+                  const addition = `to_datetime("${date.toISOString()}")${contentAfterCursor}`;
```

Comment:
Did you run into a case where you needed the explicit `to_datetime` call? I would expect implicit casting to work on a bare string.

Another question: if the explicit cast _is_ needed, do we want to use `::datetime`? Just a matter of which we want to promote as the "standard" way.

Reply:
yes it is needed unfortunately. I chose the function as I prefer it but is my personal preference 🙈 so I dont have a strong opinion here. But maybe as we are using the cast for the filtering maybe we shoudl use the same here?